feat: Mobile release process (#11506)
getsentry/sentry-docs: docs/platforms/android/releases/index.mdx

---

> title: Release Channels

Reviewer: We've adapted the naming of how Google names [their Dart/Flutter releases](https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-build-release-channels.md)

I'm not opposed to changing it to but it's already an established naming (at least in one part of the mobile world) so something to consider as well

> title: Release Channels

Reviewer 2: I think "Statuses" does not convey the actual meaning of what is being described

> title: Release Channels

Author: @lizokm thanks for the call-out! I agree that we need to make sure it's not mistaken for the Releases/Release Health product. As @buenaflor and @brustolin point out, Release Channels is a somewhat established term for what we're describing here which is why I chose it. Wdyt about changing it to `SDK Release Channels`?

> title: Release Channels

Author: @philipphofmann nope, only specific Mobile SDKs. that said, ppl are viewing this page as part of these Mobile SDK platform's docs, so feels like that would be a bit verbose, and it might be too long for the menu. but no strong opinion against it

> title: Release Channels

Reviewer: imo at least a search engine result should reflect that it's mobile only

e.g when a user searches for sentry sdk release channels it should show `Mobile SDK Release Channels` instead of only `SDK Release Channels` 

but not sure where this change would need to be added to affect those results

> title: Release Channels

Author: renamed to `Mobile SDK Releases` - wdyt?

> description: "Learn about the SDK's release channels and choose the right one."

Author: Since this change is only applicable to (specific, i.e. the "big") Mobile SDKs I didn't want to make it sound like this will be used across all of Sentry

sidebar_order: 7000
---

> This SDK uses the following ways to release SDK updates:

Reviewer 3: ```suggestion
The Android SDK has the following release statuses for SDK updates:
```
It's good for search engine optimization to make this Android-specific

## `Pre-release`

> We create pre-releases (alpha, beta, RC,…) for larger and potentially more impactful changes, such as new features or major versions.

Reviewer 3: ```suggestion
Pre-releases (alpha, beta, RC,…) are for larger and potentially more impactful changes, such as new features or major versions.
```

## `Latest`

> We continuously release major/minor/hotfix versions from the `main` branch. These releases go through all our internal quality gates and are very safe to use and intended to be the default for most teams.

Reviewer 3: ```suggestion
We continuously release major/minor/hotfix versions from the `main` branch. These releases go through all our internal quality gates and are very safe to use. They're intended to be the default for most teams.
```

## `Stable`

> We promote releases from `Latest` when they have been used in the field for some time and in scale, considering time since release, adoption, and other quality and stability metrics. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` suffix.

Reviewer 3: ```suggestion
Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` label.
```

> Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` label.

Reviewer 4: ```suggestion
Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-java/releases/) with the `Stable` label.
```

## FAQ

### Where can I find the latest stable release for any given SDK?

> On the releases pages of the SDK's GitHub repo. Stable releases will be indicated with the suffix `(Stable)` in the name of the release. You can also use the search box in the top right, and you will see the newest matching releases.

Reviewer 3: ```suggestion
On the releases pages of the SDK's GitHub repo. Stable releases will be indicated with the label `(Stable)` in the name of the release. You can also use the search box in the top right to search for the newest releases.
```

> ### Does this mean our other releases are not stable?

Reviewer 3: ```suggestion
### Does this mean the other releases we're using aren't stable?
```

> No. Stability and performance are the top priorities for every release. The `Stable` release channel is specifically designed as an additional way to build on that by utilising the fact that releases are proven in the wild an in large scale.

Reviewer 3: ```suggestion
No. Stability and performance are the top priorities for every release. The `Stable` release status means that the release has been tested and proven to work in the wild and at scale. 
```

### How often do we mark SDKs as stable?

> The target cadence is to provide monthly stable releases.

Reviewer 3: ```suggestion
The target cadence is to provide stable releases every month.
```

> The target cadence is to provide stable releases every month.

Reviewer 5: when every month?

Author: no fixed date. we "promote" existing releases that have been out for a while when it makes sense, and we plan to do this at least once a month, but it can depend on release schedule (e.g. prior to a new major there might be fewer stables for the current major)

> The target cadence is to provide stable releases every month.

Reviewer 4: Maybe we shouldn't even mention how often we mark them as stable. We could have situations where we mark a release as stable every week and perhaps only every two months.

Author: resolving this convo in favor of https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1791832673

Author: see https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1792180999
I believe it's good to give a rough indication of what the plan is

Author: > What about Mobile SDK Release Channels? Or are the other SDKs also applying this process?

see https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1792158408

Reviewer 4: Maybe we should weaken the sentence by changing it to 
`The target cadence is to provide stable releases roughly every month.`

I'm afraid people will expect a stable release every month and start complaining when there is none. I'm more a fan of it's ready when it's ready. I'm still for removing that part.

Author: reworded to `The stable release cycle depends on a number of factors and can vary, but the target cadence is to provide stable releases roughly every month.` to manage expectations. IMO it's valuable to have this here. if it creates confusion, we'll iterate on the wording (or the process)

> The stable release cycle depends on a number of factors and can vary, but the target cadence is to provide stable releases roughly every month.

Reviewer 4: 💯 that's better, thanks.

### Should I only use stable releases?

> This is a personal choice for you, your team, and your company. As stated earlier, every release is considered stable, but `Stable` releases have been tested longer with the tradeoff of a slightly delayed availability. This is also the reason why there isn't a pre-release for every release.

Reviewer 5: i think we need to reword this to give the reader/user some level of confidence that stable releases are dependable.  being a personal choice doesn't tell the customer that stable releases are actually stable. can we state anything around having a more consistent experience, fewer issues, and will get patched first if we see any issues?

Author: @rahulchhabria wdyt about "As stated earlier, every release is considered stable. On top of that, `Stable` releases have been used in the field for some time and at significant scale, and are the best choice for added dependability and a consistent experience. The tradeoff is a slightly delayed availability."?

For context about "will get patched first": we don't to commit to `Stable` meaning Long Term Support atm, but we will fix recent `Stable` (e.g. current major/minor version, latest `Stable` on previous major) releases on a best effort basis

Reviewer 5: how about 
> `Stable` releases have been tested in the field and at significant scale. They are intended for a dependable and consistent experience. Please note stable releases will get our newest features later than other releases.

Author: adopted with minor wording change

> ### Should every customer only use stable releases?
> 
> Every developer, team and company knows best which release channel works for them. As stated earlier, every release is considered stable, but `Stable` releases provide an extra layer of safety to make sure the SDK is working as expected, with the tradeoff of a slightly delayed availability. This is also the reason why it's there isn't a pre-release for every release.

Reviewer 3: ```suggestion
### Should I only use stable releases?

This is a personal choice for you, your team, and your company. As stated earlier, every release is considered stable, but `Stable` releases have been tested longer with the tradeoff of a slightly delayed availability. This is also the reason why there isn't a pre-release for every release.
```

### How does this relate to the release process and stability of a feature?

> This process doesn't affect feature stability and doesn't change the way features are rolled out. For example, a stable SDK release might include a new feature that's marked experimental.

Author: good point, we could make it more explicit. maybe something along the lines of

```This process doesn't affect the release process and stability of specific SDK or Product features, such as Session Replay. For example, a new feature that is still experimental might be part of a stable SDK release, and it will be opt-in and clearly marked as experimental.```

> This process is independent of stability of an individual feature and does not change the way features are rolled out. For example, a stable SDK release might still include a new feature that is marked experimental.

Reviewer 3: ```suggestion
This process doesn't affect feature stability and doesn't change the way features are rolled out. For example, a stable SDK release might include a new feature that's marked experimental.
```

> ### How does this relate to the stability of a feature?
> 
> The `Stable` release status doesn't ascertain the stability of specific SDK or Product features, such as Session Replay. For example, a new feature that is still experimental might be part of a stable SDK release, and it will be opt-in and clearly marked as experimental.

Reviewer 3: ```suggestion
### How does this relate to the stability of a feature?

A `Stable` release status doesn't guarantee the stability of an SDK or product feature. A new feature that's still experimental might be part of a stable SDK release. Any experimental features will be clearly marked and offered as an opt-in.
```
I want to make sure I'm understanding this correctly - are we saying that stable releases won't always have all Sentry features? Or are we saying that we don't guarantee that all features will be stable?

I'm assuming it's that we don't guarantee that all features will be stable b/c some will be experimental. 

If we're saying that we don't guarantee access to all features, we should say: A `Stable` release status doesn't guarantee the stability of a specific SDK or access to all product features.

Author: > I'm assuming it's that we don't guarantee that all features will be stable b/c some will be experimental.

Yes, that's what we're trying to convey here 👍
